python/dgl/graphbolt/utils/sample_utils.py

```diff
@@ -244,11 +251,14 @@ def unique_and_compact_node_pairs(
         src = src_nodes.get(ntype, default_tensor)
         unique_dst = unique_dst_nodes.get(ntype, default_tensor)
         dst = dst_nodes.get(ntype, default_tensor)
-        (
-            unique_nodes[ntype],
-            compacted_src[ntype],
-            compacted_dst[ntype],
-        ) = torch.ops.graphbolt.unique_and_compact(src, dst, unique_dst)
+        if deduplicate:
```

Comment:
this check should be added in L116 of python/dgl/graphbolt/impl/neighbor_sampler.py